transports/webrtc/src/webrtc_connection.rs

```diff
@@ -0,0 +1,196 @@
+
+        let peer_connection = api.new_peer_connection(config).await?;
+
+        // 1. OFFER
+        let offer = peer_connection.create_offer(None).await?;
+        log::debug!("OFFER: {:?}", offer.sdp);
+        peer_connection.set_local_description(offer).await?;
+
+        // 2. ANSWER
+        // Set the remote description to the predefined SDP.
+        let server_session_description =
+            crate::sdp::render_server_session_description(addr, remote_fingerprint);
+        log::debug!("ANSWER: {:?}", server_session_description);
+        let sdp = RTCSessionDescription::answer(server_session_description).unwrap();
+        // NOTE: this will start the gathering of ICE candidates
+        peer_connection.set_remote_description(sdp).await?;
+
+        Ok(Self { peer_connection })
+    }
+
+    pub async fn accept(
+        addr: SocketAddr,
+        config: RTCConfiguration,
+        udp_mux: Arc<dyn UDPMux + Send + Sync>,
+        our_fingerprint: &Fingerprint,
+        remote_ufrag: &str,
+    ) -> Result<Self, Error> {
+        // Set both ICE user and password to our fingerprint because that's what the client is
+        // expecting (see [`Self::connect`] "2. ANSWER").
+        let ufrag = our_fingerprint.to_ufrag();
+        let mut se = Self::setting_engine(udp_mux, &ufrag, addr.is_ipv4());
+        {
+            se.set_lite(true);
+            se.disable_certificate_fingerprint_verification(true);
+            // Act as a DTLS server (one which waits for a connection).
+            //
+            // NOTE: removing this seems to break DTLS setup (both sides send `ClientHello` messages,
+            // but none end up responding).
+            se.set_answering_dtls_role(DTLSRole::Server)?;
+        }
+
+        let api = APIBuilder::new().with_setting_engine(se).build();
+        let peer_connection = api.new_peer_connection(config).await?;
+
+        let client_session_description =
+            crate::sdp::render_client_session_description(addr, remote_ufrag);
+        log::debug!("OFFER: {:?}", client_session_description);
+        let sdp = RTCSessionDescription::offer(client_session_description).unwrap();
+        peer_connection.set_remote_description(sdp).await?;
+
+        let answer = peer_connection.create_answer(None).await?;
+        // Set the local description and start UDP listeners
+        // Note: this will start the gathering of ICE candidates
+        log::debug!("ANSWER: {:?}", answer.sdp);
+        peer_connection.set_local_description(answer).await?;
+
+        Ok(Self { peer_connection })
+    }
+
+    pub async fn create_initial_upgrade_data_channel(&self) -> Result<Arc<DataChannel>, Error> {
+        // Open a data channel to do Noise on top and verify the remote.
+        let data_channel = self
+            .peer_connection
+            .create_data_channel(
+                "data",
+                Some(RTCDataChannelInit {
+                    negotiated: Some(1),
```

Comment:
```suggestion
                    negotiated: Some(0),
```

As suggested in https://github.com/libp2p/specs/pull/412#discussion_r979413409. @melekes let me know in case you have any objections.